src/Components/GenreCards/GenreCard.js

```diff
@@ -0,0 +1,102 @@
+    setTechCategoryCheckbox,
+    setPhilosophyCategoryCheckbox,
+    setRomanceCategoryCheckbox,
+    setMangaCategoryCheckbox, 
+  } = useGenre()
+
+  const { genretype } = props
+  const { dispatchProductFilterOptions } = useProductAvailable()
+
+  function selectOnlyThatFilter()
+  {
+    switch(genretype)
+    {
+      case "Fiction" :   
+        setFictionCategoryCheckbox(true)
+        setThrillerCategoryCheckbox(false)
+        setTechCategoryCheckbox(false)
+        setPhilosophyCategoryCheckbox(false)
+        setRomanceCategoryCheckbox(false)
+        setMangaCategoryCheckbox(false)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_FICTION_FILTER"})
+        break;
+
+      case "Thriller" : 
+        setFictionCategoryCheckbox(false)
+        setThrillerCategoryCheckbox(true)
+        setTechCategoryCheckbox(false)
+        setPhilosophyCategoryCheckbox(false)
+        setRomanceCategoryCheckbox(false)
+        setMangaCategoryCheckbox(false)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_THRILLER_FILTER"})
+        break;
+
+      case "Tech" : 
+        setFictionCategoryCheckbox(false)
+        setThrillerCategoryCheckbox(false)
+        setTechCategoryCheckbox(true)
+        setPhilosophyCategoryCheckbox(false)
+        setRomanceCategoryCheckbox(false)
+        setMangaCategoryCheckbox(false)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_TECH_FILTER"})
+        break;
+
+      case "Philosophy" : 
+        setFictionCategoryCheckbox(false)
+        setThrillerCategoryCheckbox(false)
+        setTechCategoryCheckbox(false)
+        setPhilosophyCategoryCheckbox(true)
+        setRomanceCategoryCheckbox(false)
+        setMangaCategoryCheckbox(false)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_PHILOSOPHY_FILTER"})
+        break;
+
+      case "Romance" : 
+        setFictionCategoryCheckbox(false)
+        setThrillerCategoryCheckbox(false)
+        setTechCategoryCheckbox(false)
+        setPhilosophyCategoryCheckbox(false)
+        setRomanceCategoryCheckbox(true)
+        setMangaCategoryCheckbox(false)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_ROMANCE_FILTER"})
+        break;
+
+      case "Manga" : 
+        setFictionCategoryCheckbox(false)
+        setThrillerCategoryCheckbox(false)
+        setTechCategoryCheckbox(false)
+        setPhilosophyCategoryCheckbox(false)
+        setRomanceCategoryCheckbox(false)
+        setMangaCategoryCheckbox(true)
+        dispatchProductFilterOptions({type:"SELECT_ONLY_MANGA_FILTER"})
+        break;
+
+      default :    
+        setFictionCategoryCheckbox(true)
+        setThrillerCategoryCheckbox(true)
+        setTechCategoryCheckbox(true)
+        setPhilosophyCategoryCheckbox(true)
+        setRomanceCategoryCheckbox(true)
+        setMangaCategoryCheckbox(true)
+        break;
+    }  
+  }
+
+  return (
+    <div onClick={selectOnlyThatFilter} className='genre-card'>
+        <p>{props.genretype}</p>
+    </div>
+  )
+}
+
```

Comment:
Can't Understand what is happening in this file, i feel there is a lot of redundant code which could be improved
As well the i am not able to understand what is Genre Card?

Reply:
Right, so Genre Card is the blue square cards on Home Page, which user can use to directly open product page of only that particular genre.
As for redundant code, I am setting checkboxes to match genre selected and I can't think of any more concise way of doing it. It is not repeating, it is different for each scenario.